app/ui/client/lib/msgTyping.js

```diff
@@ -50,10 +50,6 @@ export const MsgTyping = new class {
 			return;
 		}
 		rooms[rid] = function(username, typing) {
-			const user = Meteor.users.findOne(Meteor.userId(), { fields: { name: 1, username: 1 } });
```

Comment:
this was made to prevent showing that yourself is typing, not sure what is worse, if showing yourself or showing someone with same name as yours (how often that happen?).

the ideal solution in my opinion is to change the typing event to add the user `_id`, so then you can compare if the user's `_id` that is typing is the same as yours and then decide to show or not based on that.

Reply:
Yes, the user's id can be compared instead of the name.
One another solution can be instead of comparing names(as they can be equal) why not compare usernames only.

Reply:
on current implementation if showing real names is turned on, that's what you'll receive from the "typing" stream :/

Reply:
I will look into this issue! :smile:

Reply:
> on current implementation if showing real names is turned on, that's what you'll receive from the "typing" stream :/

Yes, we would be getting names but is it needed? This means if we get a username irrespective of showing real names setting is on/off.